src/jit/valuenum.cpp

```diff
@@ -1801,69 +1801,105 @@ ValueNum ValueNumStore::VNOneForType(var_types typ)
+        GetVNFunc0Map()->Set(func, resultVN);
     }
+    return resultVN;
 }
 
+//----------------------------------------------------------------------------------------
+//  VNForFunc-1      - Returns the ValueNum associated with 'func'('arg0VN')
+//                     There is a one-to-one relationship between the ValueNum
+//                     and 'func'('arg0VN')
+//
+// Arguments:
+//    typ            - The type of the resulting ValueNum produced by 'func'
+//    func           - Any unary VNFunc
+//    arg0VN         - The ValueNum of the argument to 'func'
+//
+// Return Value:     - Returns the ValueNum associated with 'func'('arg0VN')
+//
+// Note: - This method only handles Unary operators
+//
 ValueNum ValueNumStore::VNForFunc(var_types typ, VNFunc func, ValueNum arg0VN)
 {
     assert(arg0VN == VNNormalValue(arg0VN)); // Arguments don't carry exceptions.
 
-    ValueNum      res;
-    VNDefFunc1Arg fstruct(func, arg0VN);
-
-    // Do constant-folding.
+    // Try to perform constant-folding.
     if (CanEvalForConstantArgs(func) && IsVNConstant(arg0VN))
     {
         return EvalFuncForConstantArgs(typ, func, arg0VN);
     }
 
-    if (GetVNFunc1Map()->Lookup(fstruct, &res))
+    ValueNum resultVN;
+
+    // Have we already assigned a ValueNum for 'func'('arg0VN') ?
+    //
+    VNDefFunc1Arg fstruct(func, arg0VN);
+    if (GetVNFunc1Map()->Lookup(fstruct, &resultVN))
     {
-        return res;
+        // success, we will return the matching resultVN
     }
     else
     {
-        // Otherwise, create a new VN for this application.
+        // Otherwise, Allocate a new ValueNum for 'func'('arg0VN')
+        //
         Chunk*   c                                                     = GetAllocChunk(typ, CEA_Func1);
         unsigned offsetWithinChunk                                     = c->AllocVN();
-        res                                                            = c->m_baseVN + offsetWithinChunk;
+        resultVN                                                       = c->m_baseVN + offsetWithinChunk;
         reinterpret_cast<VNDefFunc1Arg*>(c->m_defs)[offsetWithinChunk] = fstruct;
-        GetVNFunc1Map()->Set(fstruct, res);
-        return res;
+        // Record 'resultVN' in the Func1Map
+        GetVNFunc1Map()->Set(fstruct, resultVN);
     }
+    return resultVN;
 }
 
-// Windows x86 and Windows ARM/ARM64 may not define _isnanf() but they do define _isnan().
-// We will redirect the macros to these other functions if the macro is not defined for the
-// platform. This has the side effect of a possible implicit upcasting for arguments passed.
-#if (defined(_TARGET_X86_) || defined(_TARGET_ARM_) || defined(_TARGET_ARM64_)) && !defined(FEATURE_PAL)
-
-#if !defined(_isnanf)
-#define _isnanf _isnan
-#endif
-
-#endif
-
+//----------------------------------------------------------------------------------------
+//  VNForFunc-2      - Returns the ValueNum associated with 'func'('arg0VN','arg1VN')
```

Comment:
From docs perspective `VNForFunc-2` could be misleading since it could indicate function name. In reality func name is `VNForFunc` but is defined as different overload what cen be immediately seen from list of args

Reply:
@briansull 
I am raising this problem bcs I have started working on automatic generation of docs for native code using `doxygen`. First attempts look promising. Current documentation format is not supported by `doxygen` and good quality docs generation would require applying transforms which most probably could be implemented either by using `clang-doxygen` build or by sophisticated `regex`. Having to recognize `invalid` function name and match it with signature in code could be an additional problem preventing docs generation.

Reply:
How does `doxygen`  differenentiate between function overloads with the same name?

Reply:
`doxygen` uses either its own fully functional C++ parser or even better integrates with Clang and relies on Clang parser for C/C++ code understanding.

Reply:
Function overloads are displayed in a similar way as C# overloads are displayed in C# docs.